### ISSUE TITLE
How to use ImageBind to generate image or audio? 

### ISSUE THREAD

**Issue by NateDong72** [facebookresearch/ImageBind#42]
I can run the example code. But how to run the model to generate the some images and audio? 
![image](https://github.com/facebookresearch/ImageBind/assets/40257516/881881cf-de6a-48e4-ace5-60ca1d3f56bd)

**Comment by SoftologyPro** [comment]
Agreed.  How can you guys spend all that time training the model and writing the paper and setting up the demo website and not spend a few hours giving working example scripts to show us how to use it?

**Comment by echo-lalia** [comment]
I don't think the model can actually generate those things; I think it just 'translates' the information from one form to another. I think it'll have to be built into an extension for SD-WebUI or something, in order to let us play with it more easily.

**Comment by WilTay1** [comment]
> I don't think the model can actually generate those things; I think it just 'translates' the information from one form to another. I think it'll have to be built into an extension for SD-WebUI or something, in order to let us play with it more easily.

But the model can be downloaded and loaded in the script.

**Comment by ikuinen** [comment]
> I am also interested in this. Any news? Also, how can you retrieve an image based on image and audio/text? I am referring to the embedding space arithmetic examples in Figure 4 in the paper. Do you just sum the image embeddings with the audio/text embedding and perform cosine similarity with all the image embeddings and get the most similar image? Thanks!

We made a quick attempt: https://github.com/sail-sg/BindDiffusion

**Comment by Zeqiang-Lai** [comment]
See also https://github.com/Zeqiang-Lai/Anything2Image, which is implemented with diffusers.

**Comment by SoftologyPro** [comment]
> We made a quick attempt: https://github.com/sail-sg/BindDiffusion

This one is a good working example with a nice gradio interface.

**Comment by SoftologyPro** [comment]
> See also [Anything2Image](https://github.com/Zeqiang-Lai/Anything2Image) , it is implemented with Diffusers.

This works well with a nice gradio GUI interface.
